geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/collections/SizeableByteArrayList.java

```diff
@@ -102,22 +103,13 @@ public int lastIndexOf(Object o) {
   }
 
   /**
-   * @param remove in order (smallest to largest) list of indexes to remove
+   * @param indexes list of indexes to remove
    */
-  public void removeIndexes(List<Integer> remove) {
-    int removeIndex = 0;
-    int firstIndexToRemove = remove.get(0);
-    ListIterator<byte[]> iterator = listIterator(firstIndexToRemove);
-
-    // Iterates only through the indexes to remove
-    for (int i = firstIndexToRemove; i <= remove.get(remove.size() - 1); i++) {
-      byte[] element = iterator.next();
-      if (i == remove.get(removeIndex)) {
-        iterator.remove();
-        memberOverhead -= calculateByteArrayOverhead(element);
-        removeIndex++;
-      }
+  public void removeIndexes(List<Integer> indexes) {
+    if(indexes.size() > 1) {
+      indexes.sort(Comparator.reverseOrder());
     }
+    indexes.forEach(index -> this.remove((int)index));
```

Comment:
This change causes a significant performance hit, since we're using a LinkedList implementation that makes removing an element at an index O(N). Previously, we iterated the list once and removed the indexes as we found them, which is O(N); with this change, we iterate the list once for each index we want to remove, which is O(M*N).

Reply:
Ah, gotchya. I was wondering where the complexity was coming from -- performance code, I see. I'll revert. Though, I'll add an `int endIndexToRemove = remove.get(remove.size() - 1);` variable to make the for loop a bit easier to read.